pkg/csi/service/vanilla/controller_helper.go

```diff
@@ -231,3 +231,55 @@ func getVolumeManagerForVolumeID(ctx context.Context, controller *controller, vo
 	}
 	return volumeManager, nil
 }
+
+// getVCenterManagerForVolumeID returns vCenter manager for the given volumeId.
+// If multi-vcenter-csi-topology feature is disabled, legacy vCenter manager is returned
+// from `controller.manager.VCenterManager`.
+// If multi-vcenter-csi-topology feature is enabled, vCenter manager is returned from
+// `controller.managers.VCenterManager``.
+func getVCenterManagerForVolumeID(ctx context.Context, controller *controller, volumeId string,
+	volumeInfoService cnsvolumeinfo.VolumeInfoService) (vsphere.VirtualCenterManager, error) {
+	var vCenterManager vsphere.VirtualCenterManager
+	if multivCenterCSITopologyEnabled {
+		vCenterManager = controller.managers.VcenterManager
+	} else {
+		vCenterManager = controller.manager.VcenterManager
+	}
+	return vCenterManager, nil
+}
+
+// getVCenterHostForVolumeID returns VC host for the given volumeId.
+// If multi-vcenter-csi-topology feature is disabled legacy VC host is returned
+// from `controller.manager.VCenterConfig.Host`.
+// If multi-vcenter-csi-topology feature is enabled but volumeInfoService is not instantiated VC host
+// is returned from controller.managers.VCenterConfig.Host for `controller.managers.CnsConfig.Global.VCenterIP`.
+// If multi-vcenter-csi-topology feature is enabled and volumeInfoService is instantiated VC host is returned for
+// the vCenter IP mapped to the requested VolumeID from volumeInfoService.
+func getVCenterHostForVolumeID(ctx context.Context, controller *controller, volumeId string,
```

Comment:
Can you use this function in the `getVolumeManagerForVolumeID`?

Reply:
removed the function and returned vCenterHost from getVolumeManagerForVolumeID with some modifications in the function